### PR TITLE
Genesis block accounts indexing - Closes #421

### LINKED ISSUE #421: Genesis block account indexing
### Description

Make sure the genesis block account indexing is done properly.

### Motivation

- Genesis block data is important to make the full index.

### Acceptance Criteria

- Genesis accounts are available with all account data.

### Additional Information

The algorithm has to perform these operations:

1. Get list from `height=0`.
2. Extract addresses from `header.asset.accounts`.
3. Add the list to account index queue.

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #421 

### How was it solved?

- [x] `Indexing` related issues
  - [x] Index genesis block during `init`
  - [x] Index all accounts present in genesis block
  - [x] Fix for updating `setIndexReadyStatus` once blockchain is indexed
- [x] `Transactions` related bugs
  - [x] Fix missing sender address in some transactions
 - [x] `Accounts` related bugs
   - [x] Fix account indexing `indexAccountsByAddress` (Avoid overwriting `publicKey`)
   - [x] `producedBlocks` and `rewards` enabled in delegate response based on `setIndexReadyStatus`

### How was it tested?
- [x] Local
- [x] Jenkins

### CODE REVIEW

File: services/core/shared/core/compat/sdk_v5/blocks.js [hunk 1]
```diff
@@ -386,6 +387,9 @@ const init = async () => {
 		// Index genesis block first
 		await getBlocks({ height: genesisHeight });
 
+		// Index genesis accounts
+		await indexGenesisAccounts(genesisHeight);
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v5/blocks.js:391]: Let us merge the functionality into one method perhaps: `indexGenesisBlock`. This method should then fetch the genesis block and then invoke the `indexAccountsbyAddress` and ìndexTransactions`methods directly.

```suggestion
		await indexGenesisBlock(genesisHeight);
```

File: services/core/shared/core/compat/sdk_v5/blocks.js [hunk 2]
```diff
@@ -310,12 +320,15 @@ const buildIndex = async (from, to) => {
...
+		const batchToHeight = (offset + MAX_BLOCKS_LIMIT_PP) <= to
+			? (offset + MAX_BLOCKS_LIMIT_PP) : to;
+		logger.info(`Attempting to cache blocks ${batchFromHeight}-${batchToHeight}`);
 
 		let blocks;
 		do {
-			blocks = await getBlocksByHeightBetween(offset + 1, offset + MAX_BLOCKS_LIMIT_PP);
-		} while (!(blocks.length && blocks.every(block => !!block && !!block.height)));
+			blocks = await getBlocksByHeightBetween(batchFromHeight, batchToHeight);
+		} while (!(blocks.length && blocks.every(block => !!block && block.height >= 0)));
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v5/blocks.js:331]: Please consider updating `bottomHeightFromBatch` (line 338) to:
```
const bottomHeightFromBatch = sortedBlocks.length ? (sortedBlocks.shift()).height : topHeightFromBatch;
```
This handles an edge case scenario when the `sortedBlocks` array contains a single entry.

File: services/core/shared/core/compat/sdk_v5/blocks.js [hunk 3]
```diff
@@ -361,31 +374,34 @@ const indexMissingBlocks = async (fromHeight, toHeight) => {
...
 		const numBlocksIndexed = await blocksDB.count();
 		const [lastIndexedBlock] = await blocksDB.find({ sort: 'height:desc', limit: 1 });
-
+		/* eslint-enable no-await-in-loop */
 		if (numBlocksIndexed >= currentHeight && lastIndexedBlock.height >= currentHeight) {
 			setIndexReadyStatus(true);
 			return getIndexReadyStatus();
 		}
+			setIndexReadyStatus(false);
+			throw new Error('Block indexing is still processing...');
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v5/blocks.js:387]:
```suggestion
			throw new Error('Block indexing still in progress...');
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/core/shared/core/compat/sdk_v5/blocks.js (hunk 2 above)
Comment (nagdahimanshu) [reply services/core/shared/core/compat/sdk_v5/blocks.js:331]: Done!

**Review by sameersubudhi — APPROVED** [review]
Thank you for the changes! Good job 👍

**Review by MichalTuleja — APPROVED** [review]
LGTM 👍

**Pull request merged**